Sources/Proxy/Representations/SlicedGeometryRepresentationProxy/index.js

```diff
@@ -0,0 +1,142 @@
+import macro from 'vtk.js/Sources/macro';
+import vtkActor from 'vtk.js/Sources/Rendering/Core/Actor';
+import vtkCutter from 'vtk.js/Sources/Filters/Core/Cutter';
+import vtkImageMapper from 'vtk.js/Sources/Rendering/Core/ImageMapper';
+import vtkMapper from 'vtk.js/Sources/Rendering/Core/Mapper';
+import vtkPlane from 'vtk.js/Sources/Common/DataModel/Plane';
+
+import vtkAbstractRepresentationProxy from 'vtk.js/Sources/Proxy/Core/AbstractRepresentationProxy';
+
+// ----------------------------------------------------------------------------
+// vtkSlicedGeometryRepresentationProxy methods
+// ----------------------------------------------------------------------------
+
+function vtkSlicedGeometryRepresentationProxy(publicAPI, model) {
+  // Set our className
+  model.classHierarchy.push('vtkSlicedGeometryRepresentationProxy');
+
+  // Internals
+  model.plane = vtkPlane.newInstance();
+  model.cutter = vtkCutter.newInstance();
+  model.cutter.setCutFunction(model.plane);
+  model.mapper = vtkMapper.newInstance({
+    interpolateScalarsBeforeMapping: false,
+    useLookupTableScalarRange: false,
+    scalarVisibility: false,
+  });
+  model.actor = vtkActor.newInstance();
+  model.property = model.actor.getProperty();
+  model.property.setLighting(false);
+
+  // connect rendering pipeline
+  model.mapper.setInputConnection(model.cutter.getOutputPort());
+  model.actor.setMapper(model.mapper);
+  model.actors.push(model.actor);
+
+  // Keep things updated
+  model.sourceDependencies.push(model.cutter);
+
+  // Internal functions -------------------------------------------------------
+  function updateSlice(slice) {
+    model.slice = slice;
+    const n = model.plane.getNormal();
+    model.plane.setOrigin(n[0] * slice, n[1] * slice, n[2] * slice);
+    model.cutter.modified();
+  }
+
+  function updateOffset(offset) {
+    model.offset = offset;
+    const normal = model.plane.getNormal();
+    model.actor.setPosition(
+      offset * normal[0],
+      offset * normal[1],
+      offset * normal[2]
+    );
+  }
+
+  // API ----------------------------------------------------------------------
+  publicAPI.setSlice = (slice) => {
+    if (slice === model.slice || slice === undefined) {
+      return;
+    }
+    updateSlice(slice);
+    publicAPI.modified();
+  };
+
+  publicAPI.setOffset = (offset) => {
+    if (offset === model.offset || offset === undefined) {
+      return;
+    }
+    updateOffset(offset);
+    publicAPI.modified();
+  };
+
+  publicAPI.setSlicingMode = (mode) => {
```

Comment:
Where is the get?

Reply:
is it ok if I store it in model.slicingMode and just use the get macro?

Reply:
yes definitely that would be the way I would have done it.

Reply:
Also I would guard that method if the sliceMode remain the same. To prevent unnecessary modified/computation.
I know it seems useless, but VTK rely a lot on modified time and lot of code assume it will not changed if we keep setting the same value again and again. So to not get bitten by it later, you should ALWAYS do it.

Reply:
no it makes sense. I did not do it because i did not store model.slicingMode. Now that I do, I will.